rust/lance-encoding/src/decoder.rs

```diff
@@ -789,15 +789,26 @@ impl CoreFieldDecoderStrategy {
             let scheduler = self.create_primitive_scheduler(field, column_info, buffers)?;
             return Ok(scheduler);
         } else if data_type.is_binary_like() {
-            let column_info = column_infos.next().unwrap().clone();
+            let Some(column_info_next) = column_infos.next() else {
+                return Err(Error::InvalidInput {
+                    source: format!(
+                        "column_info cannot be none. the field type is {}, name is {}",
+                        field.data_type(),
+                        field.name
+                    )
+                    .into(),
+                    location: location!(),
+                });
+            };
+            let column_info = column_info_next.clone();
             // Column is blob and user is asking for binary data
             if let Some(blob_col) = Self::unwrap_blob(column_info.as_ref()) {
                 let desc_scheduler =
                     self.create_primitive_scheduler(&BLOB_DESC_LANCE_FIELD, &blob_col, buffers)?;
                 let blob_scheduler = Box::new(BlobFieldScheduler::new(desc_scheduler.into()));
                 return Ok(blob_scheduler);
             }
-            if let Some(page_info) = column_info.page_infos.first() {
+            return if let Some(page_info) = column_info.page_infos.first() {
```

Comment:
Could you remove the rest of these changes? It's unclear why they are necessary. They aren't related to the linked issue, right? Best to keep this PR focused on the original problem.